### PR TITLE
*: implement chunk rpc encoding for unistore

### LINKED ISSUE #35113: Unistore doesn't implement the chunk rpc encoding
## Enhancement

I find that the unistore just use the default encoding for coprocessor result.
It doesn't implement the chunk encoding protocol.

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #35113

Problem Summary:

### What is changed and how it works?

The old implementation just ignore the encode type of the coprocessor request, and use the default encoding for the response.
I make some tiny changes to support returning coprocessor response in chunk encoding. 

### Check List

Tests 

- [X] Unit test

Reuse all the old tests ...
Just set `CreateSession4Test` to use `default | chunk` encoding randomly.

- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: distsql/select_result.go [hunk 1]
```diff
@@ -140,7 +140,6 @@ type selectResult struct {
 	feedback     *statistics.QueryFeedback
 	partialCount int64 // number of partial results.
 	sqlType      string
-	encodeType   tipb.EncodeType
```
Comment (tiancaiamao) [inline distsql/select_result.go:143]: There is already a `encodeType` in `this.selectResp.DAGRequest`
This field is duplicated...

**Review by XuHuaiyu — APPROVED** [review]

**Review by wshwsh12 — APPROVED** [review]
LGTM

**Pull request merged**